tests/ui/traits/dont-autoderef-ty-with-escaping-var.stderr

```diff
@@ -5,12 +5,10 @@ LL |     <i32 as RefFoo<i32>>::ref_foo(unknown);
    |                                   ^^^^^^^ not found in this scope
 
 error[E0277]: the trait bound `for<'a> &'a mut Vec<&'a u32>: Foo<'static, i32>` is not satisfied
-  --> $DIR/dont-autoderef-ty-with-escaping-var.rs:17:35
+  --> $DIR/dont-autoderef-ty-with-escaping-var.rs:17:6
    |
 LL |     <i32 as RefFoo<i32>>::ref_foo(unknown);
-   |     ----------------------------- ^^^^^^^ the trait `for<'a> Foo<'static, i32>` is not implemented for `&'a mut Vec<&'a u32>`
-   |     |
-   |     required by a bound introduced by this call
+   |      ^^^ the trait `for<'a> Foo<'static, i32>` is not implemented for `&'a mut Vec<&'a u32>`
    |
 note: required for `i32` to implement `RefFoo<i32>`
```

Comment:
Note that pointing to `i32` is actually correct (I spend too much time thinking it was a problem), we explain in the subsequent notes *why* the `i32` is relevant.

Reply:
Yeah, it's fine. This has to do with the peeling of derived obligation causes. I don't consider this to be a regression per se.